src/styles/index.scss

```diff
@@ -2,2 +1,61 @@
+@import './variables';
+
+.page {
   margin: 0;
+  height: 100vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $days-gap;
+  width: ($calendar-item-size + $calendar-gap) * $calendar-item-columns;
+
+  &__day {
+    box-sizing: border-box;
+    position: relative;
+    padding: 10px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    background-color: $calendar-item-bg-color;
+    border: 1px solid #000;
+    width: $calendar-item-box-size;
+    height: $calendar-item-box-size;
+    transition: all 0.5s ease 0s;
+
+    @for $number-of-days from 1 through 31 {
+      &:nth-child(#{$number-of-days})::before {
+        content: '#{$number-of-days}';
+        position: absolute;
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: $calendar-item-hover-color;
+      transform: translateY(-20px);
+    }
+  }
+}
+
+@each $day, $i in $days {
+  .calendar--start-day-#{$day} {
+    .calendar__day:first-child {
+      margin-left: ($calendar-item-box-size + $days-gap) * $i;
```

Comment:
The variable `$days-gap` is used here again. Ensure it is defined in the imported `variables` file or elsewhere in your code.